### DISCUSSION TITLE
Add support for message search in a chat

### DISCUSSION THREAD

**Discussion by caltuntas** [d99kris/nchat#120] [Ideas]
It would be good to have, being able to search messages in a chat if possible.  What I mean

![image](https://github.com/d99kris/nchat/assets/35517929/1267340c-b4c3-4ee0-8e8f-d03d2ea27810)

Thanks.

**Comment by d99kris** [comment]
Hi @caltuntas - thanks for the suggestion. I can see how it would be useful, but to use corporate speak my "bandwidth is limited" currently. But if my todo-list clears up I might look into this. 

Btw, just in case some aspiring developer contributor is reading this, I'm not open for a pull request implementing search functionality until some UI has been agreed upon, as there's no "natural" way to show search results with the current UI components (I think).

**Edit:** The workaround I personally use is to export `nchat` logs anually and send them to myself via email, so I use email search (in `nmail`) to search old messages. It's not a proper replacement for search, but at least it lets one search old stuff. I have some bash script to automate this, that I can share if you'd like a copy.

**Reply by caltuntas** [reply]
I like the tool and use it daily, so if something valuable comes to mind, I post it here before I forget. However, the 'bandwidth is limited' point is completely valid. Thanks again for your effort.

Your workaround comment also inspired me to try another workaround, which involves using the `sqlite` database directly. I think I can manage with that approach for now.

**Reply by arcxne** [reply]
Hey, @caltuntas, is it possible for you to share how you got to search for messages using the sqlite db? I've tried to do it myself previously but I can't seem to get it working properly. Additionally, does your method show the contact names alongside the messages? Thank you!

**Reply by caltuntas** [reply]
Hi @arcxne , this is the screenshot, I think you can do it . 
![image](https://github.com/d99kris/nchat/assets/35517929/55f88fa6-694e-4793-98ae-6f0c11fe0b76)

**Reply by d99kris** [reply]
Hi @caltuntas - support for searching messages in a chat has been added with 879d3d4. Use alt/opt-f to find text and alt/opt-F to find next. Note: the find function only searches locally cached messages, hope it's still useful.

**Edit:** The find function jumps to match, and in order to go back to previous location (or previous search hit) in message history, one can use `page down`.
